app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -243,7 +246,23 @@ class HomeFragment : Fragment() {
         if (requireContext().settings().pocketRecommendations) {
             lifecycleScope.async(IO) {
                 val stories = components.core.pocketStoriesService.getStories()
-                homeFragmentStore.dispatch(HomeFragmentAction.PocketArticlesChange(stories))
+
+                val categoriesMapping = mutableMapOf<String, MutableList<PocketRecommendedStory>>()
+                stories.forEach { story ->
+                    categoriesMapping[story.category] = categoriesMapping
+                        .getOrPut(story.category) { mutableListOf() }
+                        .also { it.add(story) }
+                }
+                val categories = categoriesMapping.map {
+                    PocketRecommendedStoryCategory(name = it.key, stories = it.value)
+                }
+
```

Comment:
This took a while to understand. If this is grouping the stories by category in a `PocketRecommendedStoryCategory` instance then the `groupBy` operator may make this shorter. Something like:

```Kotlin
val categories = components.core.pocketStoriesService.getStories()
    .groupBy { story -> story.category }
    .map { (category, stories) -> PocketRecommendedStoryCategory(category, stories) }
```

Reply:
That looks a lot better. Thank you!